Fixes for systemd unit

seriv opened the pull request:
Typo fixed, thanks to @yakatz. Fixed excessive restarting of the unit and provided PIDFile for systemd.

yakatz committed:
Fixed spelling of multi

seriv committed:
Do not restart on succes

seriv committed:
Systemd require pidfile for forking daemons

Added option -r <pidfile> for resmon.
If none provided it will run without creating pidfile.
If provided, it will try write it's pid into specified file.
Resmon-systemd.service should have this file as an argument for
resmon as a PIDFile option and in ExecStart option.

trisk commented on resmon:
```diff
@@ -77,7 +79,15 @@ unless($debug) {
+        close PIDFILE or die "can't close $PIDFILE";
+      }
+      exit;
+    }
```
Could fix style (spaces around operators and commas) if that matters.

Not sure about convention here, but maybe lowercase the `$PIDFILE` var so it doesn't look like the file handle `PIDFILE`, or just keep using `$config->{pidfile}`.

jsime replied:
Please use the three-arg form of `open()`, instead of the two-arg form with string interpolation, and scalar filehandles instead of barewords:

`open(my $pidfh, '>', $config->{pidfile}) or die "can't open $config->{pidfile}";`

seriv committed:
fix style

seriv commented:
Thanks, updated pull request.

seriv committed:
Fix file handle comparison

'==' can not compare file handles. We can use fileno() function to
compare numerically file numbers. Code without this returns
"connection successful" when it timeouts and $fd is "undefined".
